### PR TITLE
Change Spring starter default otlp protocol from gRPC to http/protobuf

### LINKED ISSUE #10206: Change spring boot starter default otlp protocol from grpc to http/protobuf in 2.0.0
### Is your feature request related to a problem? Please describe.

I think we should probably switch the default otlp protocol in the spring boot starter (as well as the java agent) for 2.0.0

### Describe the solution you'd like

Switch the default otlp protocol in the spring boot starter for 2.0.0

### PR DESCRIPTION
**Pull request by jeanbisutti**
Fix #10206

### CODE REVIEW

**Comment by jeanbisutti** [comment]
cc @zeitlinger

**Review by zeitlinger — APPROVED** [review]
LGTM - just small nitpick :smile:

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java [hunk 1]
```diff
@@ -39,7 +42,17 @@ static <G, H, E> E applySignalProperties(
...
 
-    boolean isHttpProtobuf = Objects.equals(protocol, OtlpConfigUtil.PROTOCOL_HTTP_PROTOBUF);
+    boolean isHttpProtobuf = !"grpc".equals(protocol);
+
+    if (protocol != null
+        && !OtlpConfigUtil.PROTOCOL_GRPC.equals(protocol)
+        && !OtlpConfigUtil.PROTOCOL_HTTP_PROTOBUF.equals(protocol)) {
+      logger.warn(
+          protocol
+              + " protocol is not managed. "
```
Comment (laurit) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java:52]: I'd rephrase this to something like `Unknown OTLP protocol '{}', using 'http/protobuf' instead.`

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java [hunk 2]
```diff
@@ -39,7 +42,17 @@ static <G, H, E> E applySignalProperties(
     G grpcBuilder = newGrpcBuilder.get();
     H httpBuilder = newHttpBuilder.get();
 
-    boolean isHttpProtobuf = Objects.equals(protocol, OtlpConfigUtil.PROTOCOL_HTTP_PROTOBUF);
+    boolean isHttpProtobuf = !"grpc".equals(protocol);
+
+    if (protocol != null
```
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java:47]: Perhaps it could be rediscuss later with the perspective of what to do will the OTel starter configurations.

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java (hunk 1 above)
Comment (jeanbisutti) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java]: Done

File: instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeanbisutti) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/exporters/otlp/OtlpExporterUtil.java:47]: I have created #10213

**Review by trask — APPROVED** [review]
thx!
